src/main/java/liquibase/ext/opensearch/database/OpenSearchConnection.java

```diff
@@ -45,10 +45,23 @@
 public class OpenSearchConnection extends AbstractNoSqlConnection {
 
     private OpenSearchClient openSearchClient;
-    private Optional<OpenSearchVersionInfo> openSearchVersion = Optional.empty();
-
-    private List<URI> uris;
-    private Properties connectionProperties;
+    private Optional<InfoResponse> openSearchInfo = Optional.empty();
```

Comment:
Why do you need `Optional`s here? This is internal class state, using plain fields is more than sufficient (plus, `Optional` overuse generates tons of "garbage" for GC).

Reply:
i'm used to using `Option` in rust which is a zero-cost abstraction. it makes for super clear code and avoids dealing with `null`.

i don't think we generate a ton of garbage for GC here: we start out with `empty` and then set it once to a non-empty value and don't change it again afterwards - so that should be fine?